From a36beca8671899012fa9d1cc81fbbbbd655dbff9 Mon Sep 17 00:00:00 2001
From: X512 <danger_mail@list.ru>
Date: Tue, 22 Jun 2021 01:39:10 +0900
Subject: app_server: turn on screen on start

Screen may be turned off if video card is not supported by firmware and boot
loader.

Change-Id: Ie60fc00da281ec3781084dd97466a68b885fde7b
---
 src/servers/app/VirtualScreen.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/servers/app/VirtualScreen.cpp b/src/servers/app/VirtualScreen.cpp
index 94a9e1e1f2..92745ef561 100644
--- a/src/servers/app/VirtualScreen.cpp
+++ b/src/servers/app/VirtualScreen.cpp
@@ -136,6 +136,10 @@ VirtualScreen::AddScreen(Screen* screen, ScreenConfigurations& configurations)
 		}
 	}
 
+	// Turn on screen if this is not yet done by BIOS
+	if (status == B_OK)
+		screen->HWInterface()->SetDPMSMode(B_DPMS_ON);
+
 	// TODO: this works only for single screen configurations
 	fDrawingEngine = screen->GetDrawingEngine();
 	fHWInterface = screen->HWInterface();
@@ -144,6 +148,7 @@ VirtualScreen::AddScreen(Screen* screen, ScreenConfigurations& configurations)
 
 	fScreenList.AddItem(item);
 
+
 	return B_OK;
 }
 
-- 
2.30.2

